apollo-backend/src/test/java/io/logz/apollo/EnvironmentsStackTest.java

```diff
@@ -0,0 +1,87 @@
+package io.logz.apollo;
+
+import io.logz.apollo.clients.ApolloTestClient;
+import io.logz.apollo.helpers.Common;
+import io.logz.apollo.helpers.ModelsGenerator;
+import io.logz.apollo.helpers.StandaloneApollo;
+import io.logz.apollo.models.EnvironmentsStack;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class EnvironmentsStackTest {
+
+    private static StandaloneApollo standaloneApollo;
+    private static ApolloTestClient apolloTestClient;
+    private static int environmentsSize;
+    private static int environmentId;
+    private static List<Integer> environments = new ArrayList<>();
+    private static EnvironmentsStack environmentsStack;
+    private static int stackId;
+    private static final String NEW_STACK_NAME = "new-stack-name";
+    private static final boolean NEW_ENABLED = false;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        standaloneApollo = StandaloneApollo.getOrCreateServer();
+        apolloTestClient = Common.signupAndLogin();
+
+        environmentsSize = (int) (Math.random() * 9 + 1);
+        for (int i = 0; i < environmentsSize; i++) {
+            environmentId = ModelsGenerator.createAndSubmitEnvironment(apolloTestClient).getId();
+            environments.add(environmentId);
+        }
+        environmentsStack = ModelsGenerator.createAndSubmitEnvironmentsStack(apolloTestClient, environments);
+        stackId = environmentsStack.getId();
+    }
+
+    @Test
+    public void testAddAndGetEnvironmentsStack() throws Exception {
+        apolloTestClient.removeEnvironmentFromStack(environmentId, stackId);
+        assertThat(apolloTestClient.getEnvironmentsStack(stackId).getEnvironments()).doesNotContain(environmentId);
+        apolloTestClient.addEnvironmentToStack(environmentId, stackId);
+        assertThat(apolloTestClient.getEnvironmentsStack(stackId).getEnvironments()).contains(environmentId);
+    }
+
+    @Test
+    public void testUpdateEnvironmentsStack() throws Exception {
+        List<Integer> newEnvironmentsList = new ArrayList<>();
+        for (int i = 0; i < environmentsSize; i++) {
+            newEnvironmentsList.add(ModelsGenerator.createAndSubmitEnvironment(apolloTestClient).getId());
```

Comment:
Same comment here.
And also, why generate a whole set of new environments? two are enough to test all scenarios